front/src/modules/rollingStock/components/RollingStockEditor/CurveSpreadsheet.tsx

```diff
@@ -30,45 +35,43 @@ const CurveSpreadsheet = ({
   isDefaultCurve,
 }: CurveSpreadsheetProps) => {
   const { t } = useTranslation('rollingstock');
+  const columns = [
+    { ...keyColumn('speed', intColumn), title: t('speed') },
+    { ...keyColumn('effort', floatColumn), title: t('effort') },
+  ];
 
-  const spreadsheetCurve = useMemo(() => {
-    const { speeds, max_efforts } = selectedCurve.curve;
-    const filledMatrix: (
-      | {
-          value: string;
-        }
-      | undefined
-    )[][] =
-      speeds && max_efforts
-        ? max_efforts.map((effort, index) => [
-            {
-              value:
-                speeds[index] !== undefined ? Math.round(msToKmh(speeds[index]!)).toString() : '',
-            },
-            // Effort needs to be displayed in kN
-            { value: effort !== undefined ? Math.round(effort / 1000).toString() : '' },
-          ])
-        : [];
-    const numberOfRows = filledMatrix.length < 8 ? 8 - filledMatrix.length : 1;
-    return filledMatrix.concat(createEmptyMatrix<CellBase<string>>(numberOfRows, 2));
-  }, [selectedCurve]);
+  const [needsSort, setNeedsSort] = useState<boolean>(false);
+
+  const handleBlur = () => {
+    setNeedsSort(true);
+  };
 
-  const updateRollingStockCurve = (e: Matrix<{ value: string }>) => {
+  const updateRollingStockCurve = (newCurve: DataSheetCurve[]) => {
     if (!selectedTractionMode || !effortCurves) return;
-    const formattedCurve = formatCurve(e);
 
+    // Sort and format the curve for display
+    const sortedSpreadsheetValues = newCurve.map((item) => ({
+      speed: item.speed === null ? '' : item.speed,
+      effort: item.effort === null ? '' : item.effort,
+    }));
```

Comment:
It's not needed anymore if you use `null` everywhere instead of `''` and `undefined`
```suggestion
```

Reply:
Done